### PR TITLE
[SALAD-23120] WebApp: Mailchimp forms - integrated

### PR DESCRIPTION
**Pull request by vitto-moz**

### CODE REVIEW

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx [hunk 1]
```diff
@@ -96,22 +112,26 @@ const _DemandMonitorPage: FunctionComponent<Props> = ({
...
+    setCurrentSubscriptionStep(SubscriptionStep.Inactive)
+  }
+
+  const handleGetNotifiedContinueButtonClick = () => {
+    setCurrentSubscriptionStep(SubscriptionStep.Subscribe)
+  }
+
+  const handleSelectedHardwareNameChange = (selectedDemandHardwareName: string) => {
+    setSelectedDemandHardwareName(selectedDemandHardwareName)
+  }
```
Comment (rners01) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx:133]: we can just pass `setSelectedDemandHardwareName`
`onSelectedHardwareNameChange={setSelectedDemandHardwareName}`

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    width: '500px',
+    '@media (max-width: 812px)': {
+      width: '100%',
+    },
+  },
+})
+
+interface Props extends WithStyles<typeof styles> {
+  demandHardwareName: string
+  className?: string
```
Comment (rners01) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx:127]: you don't use it anywhere

Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx:127]: I do 
```
  return (
    <ModalWithOverlay onCloseClick={onCloseClick}>
      <div className={classes.modalWrapper}>
        <div className={classes.saladImageWrapper}>
          <div className={classes.saladImage} />
        </div>
        <div className={classes.modalContent}>
          <h1 className={classes.title}>Get Notified Of Demand Changes</h1>
          {formHtml && (
            <div className={classes.subscriptionFormWrapper} dangerouslySetInnerHTML={{ __html: formHtml }} />
          )}
        </div>
      </div>
    </ModalWithOverlay>

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx [hunk 3]
```diff
@@ -81,7 +89,15 @@ const _DemandMonitorPage: FunctionComponent<Props> = ({
   fetchDemandedHardwarePerformanceList,
   onLoginClick,
 }) => {
-  const [isModalShown, setIsModalShown] = useState(false)
+  const [currentSubscriptionStep, setCurrentSubscriptionStep] = useState<SubscriptionStep>(SubscriptionStep.Inactive)
+  const initialSelectedDemandHardwareName = demandedHardwarePerformanceList?.[0]?.name
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx:93]: Why can it be undefined? Can we use empty array as default value and not check on undefined each time?

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx (hunk 2 above)
Comment (Maks19) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx:127]: Where do you use `className`?

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/utils.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  groupId: string
+}
+
+export const getMailchimpSubscriptionForm = ({ id, fId, groupId, name }: GetMailchimpSubscriptionFormParams) => {
+  return `<div id="mc_embed_shell">
+<link href="//cdn-images.mailchimp.com/embedcode/classic-061523.css" rel="stylesheet" type="text/css">
+<style type="text/css">
+#mc_embed_signup{background:#fff; false;clear:left; font:14px Helvetica,Arial,sans-serif; width: 100%;}
+/* Add your own Mailchimp form style overrides in your site stylesheet or in this style block.
+We recommend moving this block and the preceding CSS link to the HEAD of your HTML file. /
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/utils.ts:14]: What do you think about this recommendation? Shouldn't we move it in HTML file?

Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/utils.ts:14]: I think the recommendation intended to make styles loaded faster, but we add the form dynamically, so this one can be ignored. 
The worst scenarium: we have the form looks the following for several seconds on slow 3G internet. 

image

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+const _SubscriptionDemandChangesModal = ({ classes, demandHardwareName, onCloseClick }: Props) => {
+  useEffect(() => {
+    const subscriptionForm = document.getElementById('mc-embedded-subscribe-form') as HTMLFormElement
+    subscriptionForm.onsubmit = () => {
+      onCloseClick()
+      subscriptionForm.submit()
+    }
+  }, [onCloseClick])
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx:137]:
```suggestion
useEffect(() => {
  const handleSubmit = (event) => {
    onCloseClick()
    event.target.submit()
  }

  const subscriptionForm = document.getElementById('mc-embedded-subscribe-form') as HTMLFormElement
  subscriptionForm .addEventListener('submit', handleSubmit)

  return () => {
    subscriptionForm .removeEventListener('submit', handleSubmit)
  }
}, [onCloseClick])
```

**Review by Maks19 — APPROVED** [review]

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx (hunk 5 above)
Comment (Maks19) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/SubscriptionDemandChangesModal/SubscriptionDemandChangesModal.tsx:137]: It’s an alternative method to using submit in form, but you don’t have to use it.

**Pull request merged**